app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -71,7 +188,32 @@ class ProfileForm extends Component {
       communityConfig,
     } = this.props;
     const { profile_image_90, username, name } = this.user;
-    const { canSkip } = this.state;
+    const { canSkip, groups } = this.state;
+    let sections = [];
+
+    if (groups) {
+      sections = groups.map((group) => {
+        return (
+          <div class="onboarding-profile-sub-section">
+            <h2>{group.name}</h2>
+            {group.description && (
+              <div class="color-base-60">{group.description})</div>
+            )}
+            <div>
+              {group.profile_fields.map((field) => {
+                return field.input_type === 'check_box'
```

Comment:
refactor (maybe blocking): It's a little hard to read the nested ternaries. Consider a switch case in a function and call that function with the `field.input_type`.

Reply:
✅